mailet/icalendar/src/main/java/org/apache/james/transport/mailets/ICALToJsonAttribute.java

```diff
@@ -170,27 +169,53 @@ public void service(Mail mail) throws MessagingException {
     }
 
     private void setAttribute(Mail mail, Map<String, Calendar> calendars, Map<String, byte[]> rawCalendars) throws MessagingException {
-        Optional<String> sender = retrieveSender(mail);
+        Optional<MailAddress> sender = retrieveSender(mail);
         if (!sender.isPresent()) {
             LOGGER.info("Skipping {} because no sender and no from", mail.getName());
             return;
         }
+
+        MailAddress transportSender = sender.get();
+        MailAddress replyTo = fetchReplyTo(mail).orElse(transportSender);
+
         Map<String, byte[]> jsonsInByteForm = calendars.entrySet()
             .stream()
-            .flatMap(calendar -> toJson(calendar, rawCalendars, mail, sender.get()))
+            .flatMap(calendar -> toJson(calendar, rawCalendars, mail, transportSender, replyTo))
             .collect(Guavate.toImmutableMap(Pair::getKey, Pair::getValue));
         mail.setAttribute(new Attribute(destinationAttributeName, AttributeValue.ofAny(jsonsInByteForm)));
     }
 
-    private Stream<Pair<String, byte[]>> toJson(Map.Entry<String, Calendar> entry, Map<String, byte[]> rawCalendars, Mail mail, String sender) {
+    private Optional<MailAddress> fetchReplyTo(Mail mail) throws MessagingException {
+        return Optional.ofNullable(mail.getMessage())
+            .flatMap(Throwing.<MimeMessage, Optional<String[]>>function(mimeMessage ->
+                    Optional.ofNullable(mimeMessage.getHeader(REPLY_TO_HEADER_NAME))
+                ).sneakyThrow())
+            .filter(headers -> headers.length > 0)
+            .map(headers -> headers[0])
```

Comment:
you are taking a random reply-to header when there are several ones?

Reply:
Sure, but it is supposed to be not set or set once.